src/components/Geolocation.react.js

```diff
@@ -0,0 +1,188 @@
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The CurrentLocation component gets geolocation of the device from the web browser.  See more info here:
+ * https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API
+ */
+
+export default class Geolocation extends Component {
+    constructor(props) {
+        super(props);
+        this.success = this.success.bind(this);
+        this.error = this.error.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+    }
+
+    updatePosition() {
+        if (!navigator.geolocation) {
+            alert('Your browser does not support Geolocation');
```

Comment:
Can we reuse `this.error` here? Something like
```js
this.error({
    code: 'N/A',
    message: 'Your browser does not support Geolocation'
})
```

That way (a) we respect `props.show_alert`, and (b) `props.position_error` reflects this status.

Reply:
Oh, yes - that's much better.    I made the change and it works great.